ocaml/xapi/sm_exec.ml

```diff
@@ -323,6 +323,30 @@ let with_session sr f =
       finally (fun () -> f session_id) (fun () -> destroy_session session_id)
   )
 
+(*Ideally:
+  - this should live under [xapi_observer];
+  - use [observed_components_of] on [components] instead of checking
+   if the list is empty*)
+let is_component_enabled ~component =
+  try
+    Server_helpers.exec_with_new_task
+      (Printf.sprintf "check if component %s is enbled " component)
+      (fun __context ->
+        let observers = Db.Observer.get_all ~__context in
+        List.exists
+          (fun observer ->
+            let components =
+              Db.Observer.get_components ~__context ~self:observer
+            in
+            List.length components = 0
```

Comment:
```
            components = []
```

Reply:
why do we use `components = []` here? When the component list is empty, we know that a component is enabled?

Reply:
Solved with  this reorganizing https://github.com/xapi-project/xen-api/pull/5345/commits/443adb83c760b04f6b9b3d95285d000c48c7cd38 and this fixup https://github.com/xapi-project/xen-api/pull/5345/commits/5ba8c33688c59ee52da06b7221afa40ce825cddc.

Reply:
The original suggestion still stands :)

Reply:
I used @lindig's suggestion in the end.